app/src/main/java/de/grobox/transportr/favorites/trips/AbstractFavoritesPopupMenu.java

```diff
@@ -64,4 +71,22 @@ public boolean onMenuItemClick(MenuItem item) {
 		}
 	}
 
+	protected void addShortcut(Context context, Intent shortcutIntent, String shortcutName, int shortcutDrawable) {
+		// create launcher shortcut
+		Intent addIntent = new Intent();
+		addIntent.putExtra(EXTRA_SHORTCUT_INTENT, shortcutIntent);
+		addIntent.putExtra(EXTRA_SHORTCUT_NAME, shortcutName);
+		addIntent.putExtra(EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(context, shortcutDrawable));
+		addIntent.putExtra("duplicate", false);
+		addIntent.setAction("com.android.launcher.action.INSTALL_SHORTCUT");
+		context.sendBroadcast(addIntent);
+
+		// switch to home-screen to let the user see the new shortcut
+		Intent startMain = new Intent(Intent.ACTION_MAIN);
+		startMain.addCategory(Intent.CATEGORY_HOME);
+		startMain.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+		context.startActivity(startMain);
+	}
+
+	protected abstract Intent getShortcutIntent(Context context, String type);
```

Comment:
Why is this needed if you pass in the shortcut intent above?

Reply:
Previously, it was necessary to call `getShortcutIntent()` to get the intent and pass it to `addShortcut()`.
This is unnecessary, so I changed `addShortcut()` to call `getShortcut()` itself.

`getShortcutIntent()` is needed because each subclass has its own way of creating the actual shortcut intent.